src/pybind/mgr/rbd_support/task.py

```diff
@@ -170,6 +169,7 @@ def __init__(self, module: Any) -> None:
         with self.lock:
             self.init_task_queue()
```

Comment:
What happens if we get blocklisted while initializing the queue?  I wonder if this should be moved to `run()` so that potential `ConnectionShutdown` exceptions are handled.

Reply:
I audited other 3 handlers. I should make similar changes to mirror_snapshot_schedule and trash_purge_schedule handlers as their `init_schedule_queue()` also use module's rados client that can get blocklisted.

Reply:
Good catch!

Reply:
I've made the changes. Let me know if you think it's look okay.

Reply:
> I've made the changes. Let me know if you think it's look okay.

Sorry, I think my suggestion to move the entire `init_XYZ()` method to `run()` was stupid.  This means that we would set `self.module_ready` and, even more importantly, return from `Module.__init__()` without handlers actually being initialized in the sense that mirror snapshot scheduler for example wouldn't have `self.queue` attribute!

Let's revert these changes for now and think about whether anything extra is needed at all.  The failure mode is not that bad: if any exception is thrown during module initialization, it would lead to `HEALTH_ERR` status due to an always-on module failing to load.  If it happens to be due to blocklisting, then yes, we don't recover, but it's _very_ noticeable.

Reply:
Done